### PR TITLE
Fix panic in `okteto up` when using remote deploy

### PR DESCRIPTION
**Pull request by andreafalzetti**
# Proposed changes

Fixes: DEV-147

`okteto up` would panic if the user attempted to remote deploy their service. To fix it, it was enough to pass the `ioCtrl` which was missing.

I've added a new integration test to cover this scenario. It's a copy of the existing test based on local deploy but I think it's worth having because they use very different deploy approaches.

Additionally, in the first commit, I've added a lock to prevent an additional panic that was occurring, about concurrent map writes happening in different goroutines.

## How to validate

To reproduce the bug follow these steps:

1. Using okteto `2.25.1`
1. Clone https://github.com/okteto/movies-frontend
1. Modify the `okteto.yml` to use remote deploy:
```
deploy:
  remote: true
  commands:
    - name: Deploy Frontend
      command: helm upgrade --install movies-frontend chart --set image=${OKTETO_BUILD_FRONTEND_IMAGE}
    - kubectl apply -f code.yml
```
1. Run `okteto up`
1. Notice the panic
1. Repeat the tests with the CLI built from this branch and notice how the `okteto up` succeeds.

## CLI Quality Reminders 🔧

For both authors and reviewers:

- Scrutinize for potential regressions
- Ensure key automated tests are in place
- Build the CLI and test using the validation steps
- Assess Developer Experience impact (log messages, performances, etc)
- If too broad, consider breaking into smaller PRs
- Adhere to our [code style](https://github.com/okteto/okteto/blob/master/docs/code-style.md) and [code review](https://github.com/okteto/okteto/blob/master/docs/code-review.md) guidelines

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -289,6 +289,7 @@ jobs:
           command: |
             $env:OKTETO_PATH="$($HOME)\project\artifacts\bin\okteto.exe"
             $env:Path+=";$($HOME)\project\artifacts\bin"
+            $env:SSH_AUTH_SOCK = (Get-Command ssh-agent).Definition -replace 'ssh-agent.exe','ssh-agent.sock'
```
Comment (ifbyol) [inline .circleci/config.yml:292]: Why is needed this now? For the new tests we added?

**Review by ifbyol — APPROVED** [review]
Minor comment to get more context about the change in the circleCI config but it's not a blocker

**Review by jLopezbarb — APPROVED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (andreafalzetti) [reply .circleci/config.yml:292]: @ifbyol this is needed to make remote deploy work from CI. It's already set at line [281](https://github.com/okteto/okteto/pull/4177/files#diff-78a8a19706dbd2a4425dd72bdab0502ed7a2cef16365ab7030a5a0588927bf47R281) for the `deploy` tests. Now I am adding a test for `okteto up` + remote deploy so we also need it here.

Without this, Windows Integration tests fail in CI with an error related to `SSH_AUTH_SOCK`

Comment (ifbyol) [reply .circleci/config.yml:292]: Nice! Thanks for the context!

**Pull request merged**